### PR TITLE
8269753: Misplaced caret in PatternSyntaxException's detail message

### PR DESCRIPTION
**Pull request by igraves**
Fixes a bug where carets aren't indented correctly in PatternSyntaxException messages because tab characters are converted to spaces in their indentation.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269753](https://bugs.openjdk.java.net/browse/JDK-8269753): Misplaced caret in PatternSyntaxException's detail message

### Reviewers
 * [Pavel Rappo](https://openjdk.java.net/census#prappo) (@pavelrappo - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4906/head:pull/4906` \
`$ git checkout pull/4906`

Update a local copy of the PR: \
`$ git checkout pull/4906` \
`$ git pull https://git.openjdk.java.net/jdk pull/4906/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4906`

View PR using the GUI difftool: \
`$ git pr show -t 4906`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4906.diff">https://git.openjdk.java.net/jdk/pull/4906.diff</a>

</details>

### CODE REVIEW

**Comment by igraves** [comment]
@pavelrappo

File: src/java.base/share/classes/java/util/regex/PatternSyntaxException.java [hunk 1]
```diff
@@ -107,7 +107,9 @@ public String getMessage() {
         sb.append(pattern);
         if (index >= 0 && pattern != null && index < pattern.length()) {
             sb.append(System.lineSeparator());
-            for (int i = 0; i < index; i++) sb.append(' ');
+            for (int i = 0; i < index; i++) {
+                sb.append((pattern.charAt(i) == '\t') ? '\t' : ' ');
```
Comment (pavelrappo) [inline src/java.base/share/classes/java/util/regex/PatternSyntaxException.java:111]: In contrast with https://github.com/igraves/jdk/blob/2609dd9618dd43ea0de9abe3e3100262d09c079c/src/jdk.compiler/share/classes/com/sun/tools/javac/util/AbstractDiagnosticFormatter.java#L324, this code uses `StringBuilder.append(char)`, which might be even cleaner; good.

File: test/jdk/java/util/regex/RegExTest.java [hunk 2]
```diff
@@ -5290,4 +5291,22 @@ private static void droppedClassesWithIntersection() {
...
+        } catch (PatternSyntaxException e) {
+            var message = e.getMessage();
+            var sep = System.lineSeparator();
+            if(message.contains(sep + "\t ^")){
+                return;
+            }
+        }
+        failCount++;
+
+        report("Correct caret indentation for patterns with tabs");
```
Comment (pavelrappo) [inline test/jdk/java/util/regex/RegExTest.java:5310]: `report` will not be reached on success; this is different from how `report` is used in the rest of the file. Also: `report` seems to be stateful in that it resets the `failCount`.

File: test/jdk/java/util/regex/RegExTest.java [hunk 3]
```diff
@@ -5290,4 +5291,22 @@ private static void droppedClassesWithIntersection() {
...
+    //This test is for 8269753
+    private static void errorMessageCaretIndentation() {
+        String pattern = "\t**";
+
+        try {
+            var res = Pattern.compile(pattern);
+        } catch (PatternSyntaxException e) {
+            var message = e.getMessage();
+            var sep = System.lineSeparator();
+            if(message.contains(sep + "\t ^")){
```
Comment (pavelrappo) [inline test/jdk/java/util/regex/RegExTest.java:5304]:
```suggestion
            if (message.contains(sep + "\t ^")) {
```

**Review by pavelrappo — CHANGES_REQUESTED** [review]

**Review by pavelrappo — APPROVED** [review]

**Pull request closed without merging**